OSSM-8210: Directing outbound traffic (#84144)
openshift/openshift-docs: modules/ossm-verifying-the-egress-gateway-for-http-traffic.adoc

// This procedure is used in the following assembly:
// * gateways/ossm-getting-traffic-into-a-mesh.adoc 

:_mod-docs-content-type: PROCEDURE
[id="ossm-verifying-the-egress-gateway-for-http-traffic_{context}"]
= Verifying the egress gateway for HTTP traffic

> :context: ossm-verifying-the-egress-gateway-for-http-traffic

Reviewer: Sure. I'm free any time today or Monday.

> This procedure explains how to verify that the egress gateway routes traffic to the external host.

Author: @eoinfennessy  this still needs a tweak. Maybe something like:

This procedure explains how to direct outbound HTTP traffic through a gateway that was installed by using gateway injection with Istio APIs

or maybe

This procedure explains how to use IStio APIs to direct outbound HTTP traffic through a gateway that was installed by using gateway injection

> This procedure explains how to verify that the egress gateway routes traffic to the external host.

Reviewer: I think the latter works best.

> This procedure explains how to use Istio APIs to direct outbound HTTP traffic through a gateway that was installed by using gateway injection.

> This procedure explains how to verify that the egress gateway routes traffic to the external host.

Author: thanks, done.

.Prerequisites

* You have installed a service mesh using the {SMProductName} Operator version 3.0.

> * You have created an egress gateway using gateway injection.

Author: done.

my only hesitation is if people might be confused about the need for an egress gateway.

.Procedure

. Run the following command to create a namespace called `test-namespace`:
+
[source,terminal]
----
$ oc create namespace test-namespace
----

. Enable sidecar injection in the namespace:
+
[source,terminal]
----
$ oc label namespace test-namespace istio-injection=enabled
----

Deploy the sleep application in the namespace:
+
[source,terminal]
----
$ oc apply -f https://raw.githubusercontent.com/istio/istio/release-1.23/samples/sleep/sleep.yaml -n test-namespace

> ----

Author: I believe I made this update. Eoin you can look at the revised PR and let me know if it needs a changin'

> ----

Reviewer: Yep, that looks good, thanks.

I also meant to ask did you verify with Filip or someone else that `release-1.24` branch in the `openshift-service-mesh/repo` will be available by the time TP2 is released?

> ----

Reviewer: No, we shouldn't use yaml examples from the upstream `istio/istio` repo. It has to be changed to the `openshift-service-mesh/istio` repo.

I checked the previous PR "Getting traffic into the mesh" and the repo used in the examples has been changed to upstream `istio/istio` for some reason (I had used the `openshift-service-mesh/istio` repo in the draft). I think this should be addressed in a follow up PR.

> ----

Author: Eoin, I updated it to :

oc apply -n curl -f https://raw.githubusercontent.com/openshift-service-mesh/istio/refs/heads/master/samples/curl/curl.yaml

> ----

Author: updated. done.

. Display information about the pods in the namespace:
+
[source,terminal]
----
$ oc get pods -n test-namespace

> ----

Author: Eoin, please provide the syntax. I looked in the other PR and didn't see it, but maybe I just overlooked something.

> . Store the sleep pod in the `SOURCE_POD` environment variable:
> +
> [source,terminal]
> ----
> $ export SOURCE_POD=$(oc get pod -n test-namespace -l app=sleep -o jsonpath='{.items[0].metadata.name}')
> ----

Reviewer: Nitpick to stay consistent with the name used in "Get traffic into a mesh" procedure.

```suggestion
. Export a `CURL_POD` environment variable that has been initialised with the name of the curl pod:
+
[source,terminal]
----
$ export CURL_POD=$(oc get pod -n test-namespace -l app=curl -o jsonpath='{.items[0].metadata.name}')
----
```

Author: done
